bump to 17.03.1-rc1

vieux opened the pull request:
ping @mlaventure  @mavenugo @aaronlehmann @thaJeztah am I missing any ?

yongtang committed:
Fix issue where service healthcheck is `{}` in remote API

This fix tries to address the issue raised in 30178 where
service healthcheck is `{}` in remote API will result in
dns resolve failue.

The reason was that when service healthcheck is `{}`,
service binding was not done.

This fix fixes the issue.

An integration test has been added.

This fix fixes 30178.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>
(cherry picked from commit 8feb5c5a48eaadc1686e3b370f7ef9be128dd3cb)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

albers committed:
Improve bash completion for `docker run --cap-{add,drop}`

Docker has several capabilities enabled by default and some not.
It seems natural to follow this distinction in --cap-add and
--cap-drop.

Signed-off-by: Harald Albers <github@albersweb.de>
(cherry picked from commit 36d8b66cb9aab26bd2552083b786b7a11272d9b9)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

tonistiigi committed:
overlay2: fix unmounting layer without merge dir

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
(cherry picked from commit e1af6c8a5767e38957449a1d07e56104e317191b)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

dnephin committed:
Support customizing the default network for a stack.

Signed-off-by: Daniel Nephin <dnephin@docker.com>
(cherry picked from commit b7577dd2ad6fbb2cf21baf7ee082d2f605157b2a)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

yuexiao-wang committed:
'docker daemon' deprecation message doesn't use the new versioning scheme

Signed-off-by: yuexiao-wang <wang.yuexiao@zte.com.cn>
(cherry picked from commit 5a9cee7bb57e7da7f2793fbb290498bba0531105)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

thaJeztah committed:
do not ignore "volume in use" errors when force-delete

When using `docker volume rm -f`, all errors were ignored,
and volumes where Purged, even if they were still in
use by a container.

As a result, repeated calls to `docker volume rm -f`
actually removed the volume.

The `-f` option was implemented to ignore errors
in case a volume was already removed out-of-band
by a volume driver plugin.

This patch changes the remove function to not
ignore "volume in use" errors if `-f` is used.
Other errors are still ignored as before.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>
(cherry picked from commit 9d521a4d2fbbeec0f672b986e294fd5f1c4d2f32)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

Commit:
Windows: Cleanup HCS on restore

Signed-off-by: John Howard <jhoward@microsoft.com>

This ensures that any compute processes in HCS are cleanedup
during daemon restore. Note Windows cannot (currently) reconnect
to containers on restore.

(cherry picked from commit f59593cbd1c177fe2d5a1b1f00efe9987d25a526)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

albers committed:
Improve POSIX compatibility of bash completion

Signed-off-by: Harald Albers <github@albersweb.de>
(cherry picked from commit 76f4681b569a235f9c1838fbf083fd2b8ec684ef)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

allencloud committed:
move secret into swarm thing in swagger

Signed-off-by: allencloud <allen.sun@daocloud.io>
(cherry picked from commit 6918624985e4859e44404b06a01e86639dcaa900)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

aaronlehmann commented on CHANGELOG.md:
```diff
@@ -5,6 +5,24 @@ information on the list of deprecated flags and APIs please have a look at
+### Swarm Mode
+
+* Update swarmkit to f93948cb430facd540e3c65c606384a89b3ac40f [#31742](https://github.com/docker/docker/pull/31742)
+ * Fix Tasks with a desired state < RUNNING should count [docker/swarmkit#1980](https://github.com/docker/swarmkit/pull/1980)
```
"Scheduler now correctly considers tasks which have been assigned to a node but aren't yet running"

aaronlehmann commented on CHANGELOG.md:
```diff
@@ -5,6 +5,24 @@ information on the list of deprecated flags and APIs please have a look at
+
+### Swarm Mode
+
+* Update swarmkit to f93948cb430facd540e3c65c606384a89b3ac40f [#31742](https://github.com/docker/docker/pull/31742)
```
Let's list the fixes from this PR instead. I'd summarize them as follows:

- Allow removal of a network when only dead tasks reference it docker/swarmkit#2018
- Retry failed network allocations less aggressively docker/swarmkit#2021
- Avoid network allocation for tasks that are no longer running docker/swarmkit#2017
- Bookkeeping fixes inside network allocator allocator: Remove deleted networks from the unallocated set docker/swarmkit#2019 docker/swarmkit#2020

Or we could list the other vendor PR instead of the SwarmKit PR below. But we should be consistent, and I think listing user-facing fixes is more useful.

vieux replied:
@aaronlehmann PTAL

Commit:
Fix markdown in references for Kramdown

Signed-off-by: Misty Stanley-Jones <misty@docker.com>
(cherry picked from commit 0ce13eb0d343e81db7006fd9c3e0c95ee1c3925c)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

cyphar committed:
daemon: also ensureDefaultApparmorProfile in exec path

When 567ef8e7858c ("daemon: switch to 'ensure' workflow for AppArmor
profiles") was merged, it didn't correctly handle the exec path if
AppArmor profiles were deleted. Fix this by duplicating the
ensureDefaultApparmorProfile code in the exec code.

Fixes: 567ef8e7858c ("daemon: switch to 'ensure' workflow for AppArmor profiles")
Signed-off-by: Aleksa Sarai <asarai@suse.de>
(cherry picked from commit 790a81ea9acce318d0e037771c253951b874140b)
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

thaJeztah commented:
don't think all the "possibly cherry-pick" PR's are in there 😅 https://github.com/docker/docker/pulls?utf8=✓&q=is%3Apr%20milestone%3A17.03.1%20

aaronlehmann commented:
LGTM. Sort of surprised that we're listing both "Update swarmkit to f93948cb430facd540e3c65c606384a89b3ac40f" and the fixes contained inside that version. I would think end users would only care about the actual fixes, not what Git revision of swarmkit this happens to contain. But I guess it's okay with me.

vieux committed:
update changelog

Signed-off-by: Victor Vieux <victorvieux@gmail.com>

vieux committed:
bump tp 17.03.1

Signed-off-by: Victor Vieux <victorvieux@gmail.com>

thaJeztah commented on CHANGELOG.md:
```diff
@@ -5,6 +5,38 @@ information on the list of deprecated flags and APIs please have a look at
+ * Ignore cgroup2 mountpoints [opencontainers/runc#1266](https://github.com/opencontainers/runc/pull/1266)
+* Update containerd to 595e75c212d19a81d2b808a518fe1afc1391dad5 [#31662](https://github.com/docker/docker/pull/31662)
+ * Register healtcheck service before calling restore() [docker/containerd#609](https://github.com/docker/containerd/pull/609)
+* ensureDefaultApparmorProfile in exec path [#31773](https://github.com/docker/docker/pull/31773)
```
Perhaps something like;

     Fix `docker exec` not working after unattended upgrades that reload apparmor profiles

thaJeztah reviewed (approved):
one suggestion, but LGTM otherwise

vieux commented:
ping @mlaventure @tiborvass

mlaventure reviewed (approved):
LGTM

tiborvass commented:
LGTM
